packages/browser-wallet/src/popup/popupX/pages/Onboarding/Welcome.tsx

```diff
@@ -35,13 +56,23 @@ export default function Welcome() {
     const { t } = useTranslation('x', { keyPrefix: 'onboarding.welcome' });
     const [, setCurrentNetworkConfiguration] = useAtom(networkConfigurationAtom);
     const [{ value: acceptedTerms }, setAcceptedTerms] = useAtom(acceptedTermsAtom);
+    const [, setActivityTracking] = useAtom(acceptedActivityTrackingAtom);
     const [, setUiStyle] = useAtom(uiStyleAtom);
+    const [checkboxTerms, setCheckboxTerms] = useState(false);
+    const [checkboxTracking, setCheckboxTracking] = useState(true);
     const config = useAsyncMemo(getTermsAndConditionsConfig, undefined, []);
     const nav = useNavigate();
     const navToPassword = () => {
         const version = config?.version || acceptedTerms?.version;
         // If we didn't find a version, put in an empty version
-        setAcceptedTerms({ accepted: true, version: version || '', url: config?.url });
+        setAcceptedTerms({ accepted: checkboxTerms, version: version || '', url: config?.url });
+
+        setActivityTracking(createAnalyticsTrackingObject(checkboxTracking)).then(() => {
+            appTracker.welcomeScreen();
+            appTracker.welcomeActivityTrackingCheckBoxChecked();
+            appTracker.welcomeTermAndConditionsCheckBoxChecked();
+            appTracker.welcomeGetStartedClicked();
+        });
```

Comment:
I think I would make the entire `navToPassword` function async as currently you're exposed to race conditions, i.e. if the page you navigate to calls `appTracker` methods before the tracking state is ready. This could be avoided if you:

```ts
async function navToPassword() {
  ...

  try {
    await setActivityTracking(...);

    appTracker.welcomeScreen();
    appTracker.welcomeActivityTrackingCheckBoxChecked();
    appTracker.welcomeTermAndConditionsCheckBoxChecked();
    appTracker.welcomeGetStartedClicked();
  } finally {
    ...
    nav(...);
  }
}
```